src/main/java/school/faang/misseddays/Main.java

```diff
@@ -0,0 +1,72 @@
+package school.faang.misseddays;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    public static void main(String[] args) {
+        List<Student> studentsList = new ArrayList<>();
+
+        addStudent(studentsList, "Mark", "Lawyer", 5000);
+        addStudent(studentsList, "Alex", "Programming", 2000);
+        addStudent(studentsList, "Bob", "Programming", 2001);
+        addStudent(studentsList, "John", "Programming", 2000);
+        addStudent(studentsList, "John", "Journalism", 2000);
+        addStudent(studentsList, "Jane", "Math", 2000);
+        addStudent(studentsList, "Jack", "Programming", 2000);
+
+        deleteStudent(studentsList, "John", "Programming", 2000);
+
+        searchStudentsByFacultyAndYear(studentsList, "Programming", 2000);
+
+        Map<FacultyAndYear, List<Student>> filteredByFacultyAndYear = filterStudents(studentsList);
+
+        printAllStudents(filteredByFacultyAndYear);
+    }
+
+    public static Map<FacultyAndYear, List<Student>> filterStudents(List<Student> students) {
+        Map<FacultyAndYear, List<Student>> map = new HashMap<>();
+        for (Student student : students) {
+            FacultyAndYear key = new FacultyAndYear(student.getFaculty(), student.getYear());
+            if (!map.containsKey(key)) {
+                map.put(key, new ArrayList<>());
+            }
+            map.get(key).add(student);
+        }
+        return map;
+    }
+
+    public static void addStudent(List<Student> students, String name, String faculty, int year) {
+        students.add(new Student(name, faculty, year));
+    }
+
+    public static void deleteStudent(List<Student> students, String name, String faculty, int year) {
+        Iterator<Student> iterator = students.iterator();
+        while (iterator.hasNext()) {
```

Comment:
Решение через итератор избыточно очень и усложняет тут код. Чем тебе не угодил обычный for, который сделает ровно тоже самое, только проще?)

Reply:
Я сперва сделал через for, не компилилось, и чат джипити сказал следующее :
The code you provided contains a deleteStudent method that attempts to remove a student from the list. However, modifying a list while iterating through it can lead to a ConcurrentModificationException. И посоветовал сделать через итератор

Reply:
Маловероятно, но это действительно может вызвать ошибку ConcurrentModificationException. Так как ты сначала набиваешь новый list студентов для удаления, а потом пробуешь удалить их из основного листа. Тут это не надо делать. Ты просто проходишься через FOR по студентам, ищешь совпадение и его сразу удаляешь из листа студентов, что пришел на вход. Просто используй students.remove(student);